tap/utils.py

```diff
@@ -18,20 +16,29 @@
     Generator,
     Iterator,
     List,
+    Type,
     Literal,
     Optional,
     Tuple,
+    Protocol,
     Union,
+    TypedDict,
 )
-from typing_inspect import get_args as typing_inspect_get_args, get_origin as typing_inspect_get_origin
-
-if sys.version_info >= (3, 10):
-    from types import UnionType
+a
+from typing import get_args, get_origin as typing_get_origin
 
 NO_CHANGES_STATUS = """nothing to commit, working tree clean"""
 PRIMITIVES = (str, int, float, bool)
 PathLike = Union[str, os.PathLike]
 
+def is_literal_type(tp: type) -> bool:
+    """Returns whether the type is a literal type."""
+    return tp is Literal or getattr(tp, "__origin__", None) is Literal
+
+def get_origin(tp: type) -> type:
+    """Same as typing.get_origin but returns the type itself if the origin is None."""
+    origin = typing_get_origin(tp)
+    return origin if origin is not None else tp
```

Comment:
The pieces of code that must be added to remove typing_inspect from dependencies. There is little risk of having to modify it for future versions of python since almost everything is done in the typing module